opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1056,25 +1069,31 @@ namespace Opm {
                         const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
                         const auto& fs = intQuants.fluidState();
                         const double hydrocarbon = fs.saturation(FluidSystem::oilPhaseIdx).value() + fs.saturation(FluidSystem::gasPhaseIdx).value();
-                        hcpv[region] += pv[c] * hydrocarbon;
-                        pres[region] += pv[c] * fs.pressure(FluidSystem::oilPhaseIdx).value();
+                        const double pv =
+                            ebosSimulator_.model().dofTotalVolume(c)
+                            * ebosSimulator_.problem().porosity(c);
+                        hcpv[region] += pv * hydrocarbon;
+                        pres[region] += pv * fs.pressure(FluidSystem::oilPhaseIdx).value();
                     }
                 }
                 for (int c = 0; c < nc; ++c) {
```

Comment:
> Here you still use the old loop style and the obfuscated one from above. Why?

obfuscation lies in the eye of the beholder ;). why? because it means less lines of changed code (didn't somebody complain about this recently) and at this point, one can assume that the cache for intensive quantities is "hot". but you're probably right that it should be changed to a loop over the grid because as-is it does not work with an overlap. (the same is the case for flow_legacy as far as I can see.)